public/content/posts/performance-improvement-in-a-next.js-page/index.md

```diff
@@ -0,0 +1,121 @@
+
+[This Article](/posts/improving-images-on-fishbrain.com) from our blog also explains how we have improved the overall performance of the images.
+
+### Preload the LCP image
+
+Preloading the image inside next/head is another effective way to improve the LCP score.
+
+```jsx
+<Head>
+  <link rel="preload" href={imageUrl} as="image" />
+</Head>
+```
+
+This will ask the browser to load image upfront even before it discovers the image element.
+
+### Next.js Dynamic imports
+
+For testing a single page (url) performance, [web.dev/measure](https://web.dev/measure) is a great tool. After measuring the performance with this tool, it provides us some useful methods on how to improve our web vitals scores.
+One of method it suggests is to reduce the initial response time. For this we could use [Next.js dynamic imports](https://nextjs.org/docs/advanced-features/dynamic-import). It's a handy way to load components and libraries only when those are needed. Therefore we could use the dynamic imports when we are loading components conditionally. This will also slightly reduce the build size.
+
+```jsx
+const DynamicCatchesSection = dynamic(async () =>
+  import("./components/CatchesSection")
+);
+```
+
+```jsx
+<DynamicCatchesSection
+  title="This is title"
+  regionCode={regionCode}
+  slug={slug}
+  countryCode={countryCode}
+/>
+```
+
+### Results
+
+Before:
+![Lighthouse Performance Desktop Before](/performance.png "Lighthouse Performance Desktop Before")
+
+After:
+![Lighthouse Performance Desktop After](/performance-2.png "Lighthouse Performance Desktop After")
+As you can see performance is quite good and the LCP score in particular has improved a lot compared to the earlier version.
+
+**Note:** _The performance is tested in Lighthouse using Desktop mode. Scores can be different from time to time and for mobile devices this score will be considerably lower since the mobile has fewer resources than Desktop devices._
+
+Below are a few more things you can try to improve the overall performance of a Next.js application.
+
+### Next.js image
+
+Next.js also has provide us a good image component (next/image). [Check it out here](https://nextjs.org/docs/api-reference/next/image) They are also experimenting on a new image component - [next/future/image](https://nextjs.org/docs/api-reference/next/future/image) with the help from user's feedback. It will be good to keep an eye on this one since it might help us to improve the performance of the image even more.
+
+### Static & Incremental Static Regeneration (ISR)
+
+Next js static pages are really important since it will pre-render the page at build time. We can use static page by exporting a function called `getStaticProps`. ISR pages in Next.js has also helped us a lot in improving the overall performance of our application. You can read more about it in the [Next.js docs](https://nextjs.org/docs)
```

Comment:
```suggestion
Next.js static pages are really important since they will pre-render at build time. We can use static pages by exporting a function called `getStaticProps`. ISR pages in Next.js have also helped us a lot in improving the overall performance of our application. You can read more about it in the [Next.js docs](https://nextjs.org/docs)
```